pkg/services/ngalert/notifier/channels/wecom.go

```diff
@@ -5,40 +5,82 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"net/http"
+	"time"
 
 	"github.com/prometheus/alertmanager/template"
 	"github.com/prometheus/alertmanager/types"
+	"golang.org/x/net/context/ctxhttp"
+	"golang.org/x/sync/singleflight"
 
 	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/notifications"
 )
 
+var WeComEndpoint = "https://qyapi.weixin.qq.com"
+
+const defaultWeComChannelType = "groupRobot"
+const defaultWeComMsgType = "markdown"
```

Comment:
I think we can make the message type setting more robust using a string "enumeration" (not a real enum, of course):

```go
type WeComMsgType string
const WeComMsgTypeMarkdown = "markdown" // use these in available_channels.go too
const WeComMsgTypeText = "text"
func (mt WeComMsgType) IsValid() bool {
    return mt == WeComMsgTypeMarkdown || mt == WeComMsgTypeText
}
```

Reply:
Yes, it's better.